tests/pytorch/distributed/run_numerics.py

```diff
@@ -0,0 +1,701 @@
+#!/usr/bin/python3
+
+# Copyright (c) 2022-2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# See LICENSE for license information.
+
+import sys
+import argparse
+from functools import wraps
+
+import transformer_engine.pytorch as te
+import torch
+from torch import nn
+import torch.distributed as dist
+
+from transformer_engine.common.recipe import Format, DelayedScaling
+from run_layer_with_overlap import _compare_tensors
+
+SEQ_LEN, BATCH_SIZE = 64, 64
+HIDDEN_SIZE = 64
+NR_HEADS = 4
+FFN_HIDDEN_SIZE = 1024
+WORLD_RANK, WORLD_SIZE = None, None
+NCCL_WORLD = None
+LOSS_FN = nn.MSELoss()
+FP8 = False
+
+# Fp8 recipe setup
+fp8_format = Format.HYBRID
+fp8_recipe = DelayedScaling(fp8_format=fp8_format, amax_history_len=32, amax_compute_algo="max")
+
+
+def main(argv=None, namespace=None):
+    global WORLD_RANK, WORLD_SIZE, NCCL_WORLD, FP8
+
+    dist.init_process_group(backend="nccl")
+    NCCL_WORLD = dist.new_group(backend="nccl")
+
+    WORLD_RANK = dist.get_rank()
+    WORLD_SIZE = dist.get_world_size()
+    assert WORLD_SIZE == 2, "This test uses 2 GPUs. Run with torchrun --nproc_per_node=2."
```

Comment:
`test_numerics.py` can launch with a different number of GPUs:
https://github.com/NVIDIA/TransformerEngine/blob/fd67d8f3c3d32ba0f149ed0b0286bfd17e9db678/tests/pytorch/distributed/test_numerics.py#L32
```suggestion
```

Reply:
I changed test_numerics.py to always run with 2 processes - it makes tests simpler.